test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -1049,3 +1075,30 @@
                         (utils/get-extension-value utils/subject-alt-name-oid)
                         (:dns-name))]
       (is (= #{"puppet" "master"} (set alt-names))))))
+
+(deftest file-permissions
+  (let [tmpdir (fs/tmpdir)]
+    (testing "A newly created file contains the properly set permissions"
+      (doseq [u all-perms
+              g all-perms
+              o all-perms]
+        (let [tmp-file (str tmpdir (UUID/randomUUID) (.getTime (Date.)))
```

Comment:
We can just use `(fs/temp-name tmpdir)` instead of building the temporary name manually.

Can then get rid of the `UUID` and `Date` imports too